docs/import/add_disaster.js

```diff
@@ -0,0 +1,253 @@
+import {eeLegacyPathPrefix, gdEePathPrefix} from '../ee_paths.js';
+import {getFirestoreRoot} from '../firestore_document.js';
+
+export {enableWhenReady, SENTINEL_NEW_DISASTER_VALUE, toggleState};
+// Visible for testing
+export {
+  addDisaster,
+  createAssetPickers,
+  createOptionFrom,
+  disasters,
+  emptyCallback,
+  getAssetsFromEe,
+  stateAssets,
+  writeNewDisaster,
+};
+
+// Currently a map of disaster name to states. This pulls once on firebase
+// authentication and then makes local updates afterwards so we don't need to
+// wait on firebase writes to read new info.
+const disasters = new Map();
+
+// Map of state to list of known assets
+const stateAssets = new Map();
+
+const SENTINEL_OPTION_VALUE = '...';
+const SENTINEL_NEW_DISASTER_VALUE = 'NEW DISASTER';
+
+/**
+ * Populates the disaster picker with disasters from firestore, and enables
+ * the ability to add a new disaster.
+ * @return {Promise<firebase.firestore.QuerySnapshot>}
+ */
+function enableWhenReady() {
+  // enable (currently hidden) add disaster button now that firestore is ready.
+  const addDisasterButton = $('#add-disaster-button');
+  addDisasterButton.prop('disabled', false);
+  addDisasterButton.on('click', addDisaster);
+
+  // populate disaster picker.
+  return getFirestoreRoot()
+      .collection('disaster-metadata')
+      .get()
+      .then((querySnapshot) => {
+        const disasterPicker = $('#disaster');
+        disasterPicker.append(createOptionFrom(SENTINEL_OPTION_VALUE));
+        querySnapshot.forEach((doc) => {
+          const name = doc.id;
+          disasterPicker.append(createOptionFrom(name));
+          disasters.set(name, doc.data().states);
+        });
+        disasterPicker.on('change', () => toggleState(disasterPicker.val()));
+      });
+}
+
+/**
+ * Switch between the page creating a new disaster and the page editing a
+ * current disaster.
+ * @param {String} disaster
+ * @return {Promise<void>} completes when we've finished filling all state
+ * pickers.
+ */
+function toggleState(disaster) {
+  if (disaster === SENTINEL_NEW_DISASTER_VALUE) {
+    $('#new-disaster').show();
+    $('#selected-disaster').hide();
+  } else if (disaster === SENTINEL_OPTION_VALUE) {
+    $('#new-disaster').hide();
+    $('#selected-disaster').hide();
+  } else {
+    const states = disasters.get(disaster);
+    const statesToFetch = [];
+    for (const state of states) {
+      if (!stateAssets.has(state)) statesToFetch.push(state);
+    }
+
+    // TODO: add functionality to re-pull all cached states from ee without
+    // reloading the page.
+    const assetPickersDone = getAssetsFromEe(statesToFetch).then((assets) => {
```

Comment:
should probably avoid this call if statesToFetch is empty

Reply:
sgtm